### PR TITLE
fix: prevent app from crashing in dev mode when yaml has undefined values

### LINKED ISSUE #176: Investigate failing parsing YAML on local dev environment
Context:
Running locally via `yarn dev`, with a wallet connected to mainnet.

The `Govern` tab doesn't load, and displays an error instead:
image

Specifically, [CGP 99](https://github.com/celo-org/governance/blob/main/CGPs/cgp-0099.md) triggers the problem. 
This proposal has no value (or placeholder) for date-executed. When this proposal is omitted from the list, the page loads without error.

In the function `parsedYAMLFromMarkdown` defined in src/utils/proposals.ts, it assumes 2-element arrays resulting from splitting a string on a `:`. For CGP 99, this array only contains a single element, and doesn't fail gracefully in a dev environment.

It seems to work in production.

### PR DESCRIPTION
**Pull request by nicolasbrugneaux**
Also replace the handcrafted yaml parser by [gray-matter](https://www.npmjs.com/package/gray-matter)

[Closes #176]

### CODE REVIEW

File: src/features/governance/data/getProposals.ts [hunk 1]
```diff
@@ -142,6 +142,7 @@ export async function getYamlForProposal(proposal: MiniProposal) {
     if (!parsed) {
       throw new Error("Couldn't parse markdown");
     }
+    console.log(parsed);
```
Comment (yerdua) [inline src/features/governance/data/getProposals.ts:145]: this was just for debugging, right?

**Review by yerdua — APPROVED** [review]
Aside from a stray debugging console.log statement, this looks good to go.

File: src/features/governance/data/getProposals.ts (hunk 1 above)
Comment (nicolasbrugneaux) [reply src/features/governance/data/getProposals.ts:145]:
```suggestion
-    console.log(parsed);
```

Comment (nicolasbrugneaux) [reply src/features/governance/data/getProposals.ts:145]:
```suggestion
```

**Pull request merged**